### ISSUE TITLE
Recursive model RuntimeError

### ISSUE THREAD

**Issue by dmfigol** [samuelcolvin/pydantic#251]
* OS: **macOS**
* Python version `import sys; print(sys.version)`: **3.6.5**
* Pydantic version `import pydantic; print(pydantic.VERSION)`: **0.13**

I am trying to represent the following data:

```
data = [        
        {
            "templateId": "b4681fcf-458d-4d59-8bb7-bab08b3484d9",
            "templateType": "type1"
        },
        {
            "templateId": "6506d576-969c-4c64-9497-f45357fa12e0",
            "templateType": "type2",
            "subTemplates": [
                {
                    "templateId": "5b300fe2-e430-46b6-8c35-04f8cec09af3",
                    "templateType": "type100"
                }
            ]
        }
]
```
Here is my model:
```py
from typing import List, Dict, Any, Optional

from pydantic import BaseModel, Schema, validator


class AttachedTemplate(BaseModel):
    id: str = Schema(..., alias="templateId")
    type: str = Schema(..., alias="templateType")
    subtemplates: Optional[List["AttachedTemplate"]] = Schema(None, alias="subTemplates")
```
It crashes during model definition:
```
...
../python3.6/site-packages/pydantic/fields.py in _create_sub_type(self, type_, name)
    255             required=self.required,
    256             allow_none=self.allow_none,
--> 257             model_config=self.model_config,
    258         )
    259

../python3.6/site-packages/pydantic/fields.py in __init__(self, name, type_, class_validators, default, required, model_config, alias, allow_none, schema)
     85         self.shape: Shape = Shape.SINGLETON
     86         self._schema: Schema = schema
---> 87         self.prepare()
     88
     89     @classmethod

../python3.6/site-packages/pydantic/fields.py in prepare(self)
    134
    135         self._populate_sub_fields()
--> 136         self._populate_validators()
    137
    138     def schema(self, by_alias=True):

../python3.6/site-packages/pydantic/fields.py in _populate_validators(self)
    264                 *tuple(v.func for v in self.class_validators if not v.whole and v.pre),
    265                 *(get_validators() if get_validators else find_validators(self.type_,
--> 266                                                                           self.model_config.arbitrary_types_allowed)),
    267                 *tuple(v.func for v in self.class_validators if not v.whole and not v.pre),
    268             )

../python3.6/site-packages/pydantic/validators.py in find_validators(type_, arbitrary_types_allowed)
    263                 return validators
    264         except TypeError as e:
--> 265             raise RuntimeError(f'error checking inheritance of {type_!r} (type: {display_as_type(type_)})') from e
    266
    267     if arbitrary_types_allowed:

RuntimeError: error checking inheritance of _ForwardRef('AttachedTemplate') (type: _ForwardRef('AttachedTemplate'))
```

**Comment by samuelcolvin** [comment]
are you sure this isn't the same as #234?

**Comment by samuelcolvin** [comment]
and could be fixed by #248?

**Comment by dmfigol** [comment]
It looks like #234 indeed.
